lib/functions/ProofUtil.ts

```diff
@@ -35,25 +43,34 @@ function partiallyValidateJWS(jws: string): void {
   }
 }
 
-function setJWSDefaults(args: JWTSignerArgs, issuerUrl: string, clientId?: string): JWTSignerArgs {
+function setJWSDefaults(args: ProofOfPossessionCallbackArgs, endpointMetadata: EndpointMetadata): { header: JWTHeader; payload: JWTPayload } {
   const now = +new Date();
-  const aud = args.payload.aud ? args.payload.aud : issuerUrl;
+  if (!endpointMetadata) {
+    throw new Error('No endpointMetadata provided');
+  }
+  const aud = endpointMetadata.issuer ? endpointMetadata.issuer : args.payload.aud;
   if (!aud) {
     throw new Error('No issuer url provided');
   }
-  const iss = args.payload.iss ? args.payload.iss : clientId;
+  if (!args.kid) {
+    throw new Error('No kid provided');
+  }
+  const iss = endpointMetadata.token_endpoint ? endpointMetadata.token_endpoint : args.payload.iss;
```

Comment:
This doesn't make sense. this is the client id that needs to be passed in. It has nothing to do with the token endpoint. 

The client is the issuer/iss of the PoP JWT. That has nothing todo with the credential issuer. The client is creating a signed message that proves they are in control of the keys associated with the kid value

Reply:
Added another field to builder for `clientId`